unbalanced_dataset/over_sampling/smote.py

```diff
@@ -383,6 +383,10 @@ def transform(self, X, y):
         else:
             num_samples = ((self.ratio_ * self.stats_c_[self.maj_c_]) -
                            self.stats_c_[self.min_c_])
+        
+        if num_samples <= 0:
+            print("Warning: with current ratio %f downsampling would be needed! Returning original X, y."%self.ratio_)
```

Comment:
Could you raise a proper warning:

https://docs.python.org/2/library/warnings.html

You should add a test to check this branch for the coverage.